### PR TITLE
Add Rich Nav scopes for Find All References

### PR DESCRIPTION
**Pull request by AmadeusW**
Adds 3 new scopes to Find All References:
"Entire Organization"
"Entire Repository"
"Loaded Solution" (replaces "Entire Solution")

### CODE REVIEW

**Comment by AmadeusW** [comment]
I'm seeing an error which I don't know how to fix
```D:\src\roslyn\src\Setup\DevDivInsertionFiles\DevDivInsertionFiles.csproj(145,5): error : Unexpected dependencies found -- update Expect edDependency list: MessagePack.Annotations;MessagePack;Microsoft.Diagnostics.Tracing.EventSource.Redist;Microsoft.ServiceHub.Client;Mic rosoft.ServiceHub.Framework;Microsoft.ServiceHub.Resources```
Note that I needed to update the Shell dependency to be able to export new scopes.

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+        public IErrorListFilterHandler ReplaceFilter(IWpfTableControl tableControl, string filterIdentifier)
+        {
+            if (filterIdentifier == PredefinedScopeFilterNames.AllItemsScopeFilter)
+            {
+                return new LoadedSolutionFilterHandler("Loaded Items");
+            }
+            else if (filterIdentifier == PredefinedScopeFilterNames.EntireSolutionScopeFilter)
+            {
+                return new LoadedSolutionFilterHandler("Loaded Solution");
+            }
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:50]: i'm curious why it's done this way where this ScopeFilterFactory handles two different filter identifiers.

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+            return null; // Don't replace
+        }
+    }
+
+    internal class LoadedSolutionFilterHandler : FilterHandlerBase
+    {
+        private readonly string displayName;
+        internal const int LoadedSolutionFilterHandlerFilterId = 20;
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:58]: are the actual constant values of these relevant?  or is this just for ordering?

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        {
+            // We're only replacing an existing filter, and not creating a new one.
+            return null;
+        }
+
+        public IErrorListFilterHandler ReplaceFilter(IWpfTableControl tableControl, string filterIdentifier)
+        {
+            if (filterIdentifier == PredefinedScopeFilterNames.AllItemsScopeFilter)
+            {
+                return new LoadedSolutionFilterHandler("Loaded Items");
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:45]: expectation is that these would be localized by roslyn right?

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        internal ItemOriginFilter(ItemOrigin targetOrigin)
+        {
+            this.targetOrigin = targetOrigin;
+        }
+
+        public bool Match(ITableEntryHandle entry)
+        {
+            Requires.NotNull(entry, nameof(entry));
+
+            if (entry.TryGetValue(StandardTableKeyNames.ItemOrigin, out ItemOrigin entryOrigin))
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:183]: so this works by grabbing an ITemOrigin out of hte entry?  i assume this means i should be adding that ItemOrigin to the entry based on the values i get from richnav?    Is there a set of well-known origins?

Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:183]: also, i don't think rich nav provides this info.  should i just assume all the results i get back for searching a particular scope *are* in that scope?

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs (hunk 2 above)
Comment (AmadeusW) [reply src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:58]: They are relevant for programmatic selection. It's a part of a very old and internal API.
These values must be different for each handler. I guarantee that values in 20s are free to use.

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs (hunk 3 above)
Comment (AmadeusW) [reply src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:45]: They would be localized by whoever owns this code. I guess we have to localize it now, even though it's temporary.

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs (hunk 4 above)
Comment (AmadeusW) [reply src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:183]: RichNav is providing this info. Note that when searching in Organization scope, you might be also getting results specific from your Repository. They would have a different scope

Comment (AmadeusW) [reply src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:183]: The key names are in the `Microsoft.VisualStudio.Shell.Framework` dll, in `Microsoft.VisualStudio.Shell.TableManager.StandardTableKeyNames` type
image

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs (hunk 1 above)
Comment (AmadeusW) [reply src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:50]: This factory provides instances of `LoadedSolutionFilterHandler`. It happens that this handler is an appropriate replacement for both `AllItemsScopeFilter` and `EntireSolutionScopeFilter`

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:45]: that would be me.  just checking :)

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:50]: what is teh 'all items' scope?

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs (hunk 2 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:58]: great.  so these are magic, and i can mark them as such :)

File: src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs (hunk 4 above)
Comment (CyrusNajmabadi) [reply src/VisualStudio/Core/Def/Implementation/FindUsages/RichNavFilterScopes.cs:183]: > RichNav is providing this info.

Do you know where?  All they pass us back is a LocationWithText.  Does that have Origin data in it?

**Pull request closed without merging**